File: mobile/lib/src/view_models/attachment.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'attachment.freezed.dart';

@freezed
abstract class Attachment with _$Attachment {
  const factory Attachment({
    required Uri uri,
    required String mimeType,
    required String fileName,
  }) = _Attachment;
}

File: backend/LeanCode.Chat/Services/DataAccess/Entities/Attachment.cs
using System;

namespace LeanCode.Chat.Services.DataAccess.Entities;

public class Attachment
{
    public Uri Uri { get; private set; } = null!;
    public string MimeType { get; private set; } = null!;
    public string FileName { get; private set; } = null!;

    private Attachment() { }

    public Attachment(Uri uri, string mimeType, string fileName)
    {
        Uri = uri;
        MimeType = mimeType;
        FileName = fileName;
    }
}
